packages/tupaia-web/src/Routes.tsx

```diff
@@ -52,6 +53,15 @@ const UserPageRedirect = ({ modal }: { modal: MODAL_ROUTES }) => {
  *
  * **/
 export const Routes = () => {
+   const [urlSearchParams] = useSearchParams();
+   const { mutate: attemptLogin } = useOneTimeLogin();
+   const token = urlSearchParams.get(URL_SEARCH_PARAMS.ONE_TIME_LOGIN_TOKEN);
```

Comment:
I'd probably suggest moving this into it's own custom hook, and then calling the custom hook from here. Just because there are a few lines of logic but all related to this one thing, and it keeps it contained